gapis/gfxapi/gles/gles.api

```diff
@@ -515,34 +515,47 @@ sub void ApplyStaticContextState(ref!Context ctx, ref!StaticContextState staticS
 
 sub void ApplyDynamicContextState(ref!Context ctx, ref!DynamicContextState dynamicState) {
   if (dynamicState != null) {
-    backbuffer := ctx.Objects.Framebuffers[0]
-    color_id := backbuffer.ColorAttachments[0].Renderbuffer.ID
-    depth_id := backbuffer.DepthAttachment.Renderbuffer.ID
-    stencil_id := backbuffer.StencilAttachment.Renderbuffer.ID
-
-    color_buffer := ctx.SharedObjects.Renderbuffers[color_id]
-    color_buffer.Width = dynamicState.BackbufferWidth
-    color_buffer.Height = dynamicState.BackbufferHeight
-    color_buffer.InternalFormat = dynamicState.BackbufferColorFmt
-    ctx.SharedObjects.Renderbuffers[color_id] = color_buffer
-
-    depth_buffer := ctx.SharedObjects.Renderbuffers[depth_id]
-    depth_buffer.Width = dynamicState.BackbufferWidth
-    depth_buffer.Height = dynamicState.BackbufferHeight
-    depth_buffer.InternalFormat = dynamicState.BackbufferDepthFmt
-    ctx.SharedObjects.Renderbuffers[depth_id] = depth_buffer
-
-    stencil_buffer := ctx.SharedObjects.Renderbuffers[stencil_id]
-    stencil_buffer.Width = dynamicState.BackbufferWidth
-    stencil_buffer.Height = dynamicState.BackbufferHeight
-    stencil_buffer.InternalFormat = dynamicState.BackbufferStencilFmt
-    ctx.SharedObjects.Renderbuffers[stencil_id] = stencil_buffer
-
-    if dynamicState.ResetViewportScissor {
-      ctx.FragmentOperations.Scissor.Box.Width = dynamicState.BackbufferWidth
-      ctx.FragmentOperations.Scissor.Box.Height = dynamicState.BackbufferHeight
-      ctx.Rasterization.Viewport.Width = dynamicState.BackbufferWidth
-      ctx.Rasterization.Viewport.Height = dynamicState.BackbufferHeight
+    if 0 in ctx.Objects.Framebuffers {
```

Comment:
I am not happy about this - I do want to write the whole API file based on the assumption the data might get inconsistent at any point.  This is just one case, but there is still hundreds left all over the API file, and all over the compat layers, etc... 

If the state might get corrupt, we need to fix the underlying cause - which is context cancellation in this case.  I still maintain my original position that context cancellation is generally bad idea - it is just a trigger that will make your code start failing at random point under hard to reproduce conditions.  If you kick of a job, the job should finish.  It is fine for very expensive jobs to have high level check if the context has been cancelled - so the top most loop of replay could bail out early.  But having leaf sub systems fail and then trying to deal with the fallout is just asking for trouble.

Having said that, we had this discussion many time before, so I am happy with you keeping the cancellation policy as it is, as long as you are the one who deals who plays whack-a-mole with the resulting problems :-)